MekHQ/src/mekhq/campaign/enums/PlanetaryAcquisitionFactionLimit.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2021 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+package mekhq.campaign.enums;
+
+import megamek.common.util.EncodeControl;
+import mekhq.MekHQ;
+
+import java.util.ResourceBundle;
+
+public enum PlanetaryAcquisitionFactionLimit {
+    //region Enum Declarations
+    ALL("PlanetaryAcquisitionFactionLimit.ALL.text"),
+    NEUTRAL("PlanetaryAcquisitionFactionLimit.NEUTRAL.text"),
+    ALLIED("PlanetaryAcquisitionFactionLimit.ALLIED.text"),
+    SELF("PlanetaryAcquisitionFactionLimit.SELF.text");
+    //endregion Enum Declarations
+
+    //region Variable Declarations
+    private final String name;
+
+    private final ResourceBundle resources = ResourceBundle.getBundle("mekhq.resources.Campaign", new EncodeControl());
+    //endregion Variable Declarations
+
+    //region Constructors
+    PlanetaryAcquisitionFactionLimit(String name) {
+        this.name = resources.getString(name);
+    }
+    //endregion Constructors
+
+    //region Boolean Comparison Methods
+    public boolean isAll() {
+        return this == ALL;
+    }
+
+    public boolean isNeutral() {
+        return this == NEUTRAL;
+    }
+
+    public boolean isAllied() {
+        return this == ALLIED;
+    }
+
+    public boolean isSelf() {
+        return this == SELF;
+    }
+
+    public boolean generateEnemy() {
+        return isAll();
+    }
+
+    public boolean generateNeutral() {
+        return generateEnemy() || isNeutral();
```

Comment:
Should this instead be `isAll() || isNeutral()`? Seems off that asking to generate enemies would also bring along neutrals (not speaking to the logic in question, just the words/concepts used).

Reply:
The logic is that if you allow planetary acquisitions from an enemy-held planet, you also allow acquisitions from neutral planets. This carries forwards so that if you allow generation on a neutral planet, you also allow it on allied planets. Maybe changing the name to generateOnNeutralPlanets would be better?

Reply:
I've changed the names of the methods in question. The logic is the same, but naming it as generateOn{{Type}}Planet should prevent the confusion in future.